src/hooks/useData.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import { IMovie } from '../shared/types';
+
+const useMovies = (
+  searchQuery: string | undefined,
+  activeGenre: string | undefined,
+  sortBy: string
+) => {
+  const [responseData, setResponseData] = useState<IMovie[]>([]);
+
+  useEffect(() => {
+    let filter = '';
+    if (activeGenre !== 'All') {
```

Comment:
It's better to extract all strings that you're using more than one time into constants to avoid typos